### PR TITLE
Draft: middleware settings approach

### PR DESCRIPTION
**Pull request by rimrakhimov**

### CODE REVIEW

File: smart-contract-verifier/smart-contract-verifier-http/src/settings.rs [hunk 1]
```diff
@@ -219,6 +221,25 @@ impl Default for CompilersSettings {
     }
 }
 
+#[serde_as]
+#[derive(Debug, Clone, Default, PartialEq, Eq, Deserialize)]
+#[serde(default, deny_unknown_fields)]
+pub struct ExtensionsSettings {
+    #[serde_as(as = "serde_with::EnumMap")]
+    solidity: Vec<Extensions>,
```
Comment (leviska) [inline smart-contract-verifier/smart-contract-verifier-http/src/settings.rs:229]: do we need a vec of extensions? Can't we make this a struct? like
```rust
struct Extensions {
  sig_provider: sig_provider::Settings,
  sourcify: sourcify::Settings,
}
```

Comment (leviska) [reply smart-contract-verifier/smart-contract-verifier-http/src/settings.rs:229]: I think we don't need to abstract this out. It's good to see a list of supported extensions

Comment (rimrakhimov) [reply smart-contract-verifier/smart-contract-verifier-http/src/settings.rs:229]: We've discussed that question in private conversation, and come up that the struct could be used as well. I personally have no preference in that case. @leviska I believe it's up to you to choose whatever you find more appropriate )

**Comment by leviska** [comment]
Implemented in #217

**Pull request closed without merging**